components/formats-gpl/src/loci/formats/in/KLBReader.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * #%L
+ * OME Bio-Formats package for reading and converting biological file formats.
+ * %%
+ * Copyright (C) 2005 - 2018 Open Microscopy Environment:
+ *   - Board of Regents of the University of Wisconsin-Madison
+ *   - Glencoe Software, Inc.
+ *   - University of Dundee
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the 
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public 
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+package loci.formats.in;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import loci.common.CBZip2InputStream;
+import loci.common.RandomAccessInputStream;
+import loci.formats.CoreMetadata;
+import loci.formats.FormatException;
+import loci.formats.FormatReader;
+import loci.formats.FormatTools;
+import loci.formats.codec.CodecOptions;
+import loci.formats.codec.ZlibCodec;
+import loci.formats.meta.MetadataStore;
+
+import ome.units.UNITS;
+
+import ome.xml.model.primitives.PositiveFloat;
+
+/**
+ * Reader for Keller Lab Block (KLB) files.
+ */
+public class KLBReader extends FormatReader {
+
+  // -- Constants --
+  private static final int KLB_DATA_DIMS = 5; //images have at most 5 dimensions: x,y,z, c, t
+  private static final int KLB_METADATA_SIZE = 256; //number of bytes in metadata
+  private static final int KLB_DEFAULT_HEADER_VERSION = 2;
+
+  private static final int UINT8_TYPE = 0;
```

Comment:
Seeing as we're introducing new code here would it be okay to use enums for this kind of thing?